monkey/infection_monkey/island_api_client/configuration_validator_decorator.py

```diff
@@ -25,8 +25,8 @@ class ConfigurationValidatorDecorator(IIslandAPIClient):
     def __init__(self, island_api_client: IIslandAPIClient):
         self._island_api_client = island_api_client
 
-    def connect(self, island_server: SocketAddress):
-        return self._island_api_client.connect(island_server)
+    def login(self, otp: str):
```

Comment:
Off-topic, but isn't there something we could do to not have to thin-wrap all of the IIslandAPIClient methods? Most of them are irrelevant for this decorator... Maybe we could just override the methods in the constructor and return an instance of `IIslandAPIClient`? I remember we looked into it but maybe 3.11 has something better?

Reply:
We can investigate if 3.11 has something new we can use. I think we looked into this with 3.7 and hit walls.